### PR TITLE
W-11331265 shared tenancy

### PR DESCRIPTION
**Pull request by IsaacEldridge**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/install-self-managed.adoc [hunk 1]
```diff
@@ -15,9 +15,10 @@ Before installing Anypoint Runtime Fabric in a Kubernetes environment, ensure:
 ** An ingress controller to send external requests to applications.
 * You create your clusters using the latest Runtime Fabric agent version available. If you can't do so, refer to https://help.mulesoft.com/s/article/RTF-Agent-upgrade-path[How to Specify the RTF Agent Version in Upgrades^].
 
-[WARNING]
+[IMPORTANT]
 ====
-If you want to use a local registry with Runtime Fabric, follow the installation instructions in xref:configure-local-registry.adoc[Using a Local Registry with Runtime Fabric]. You must set up the local registry during installation.
+* If you want to use a local registry with Runtime Fabric, follow the installation instructions in xref:configure-local-registry.adoc[Using a Local Registry with Runtime Fabric]. You must set up the local registry during installation.
+* If you want to set up shared tenancy, you must configure it before installing Runtime Fabric. See xref:shared-tenancy.adoc[Configuring Shared Tenancy in Runtime Fabric] for instructions.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/install-self-managed.adoc]:
```suggestion
* If you want to set up shared tenancy, configure it before installing Runtime Fabric. See xref:shared-tenancy.adoc[Configuring Shared Tenancy in Runtime Fabric] for instructions. 
```

File: modules/ROOT/pages/shared-tenancy.adoc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+== How Shared Tenancy Works
+
+At install time, you provide a configuration file that consists of your namespaces, ServiceAccounts, ClusterRoles/Roles, and RoleBindings.
+
+Runtime Fabric installs the rtf-agent in the Runtime Fabric namespace, and it has access to any additional namespaces you create.
+
+=== Example
+
+The following is an example of a rtf:agent ClusterRole that allows everything regarding cronjobs. available in the whole cluster, but doesn’t itself grant permissions for the whole cluster. It’s available in the whole cluster, but doesn’t itself grant permissions for the whole cluster.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/shared-tenancy.adoc]: This paragraph is confusing and repeats text. What I'm understanding from this is:

The following is an example of a _rtf:agent ClusterRole_ that allows every action/command regarding _cronjobs_ that are available in the cluster but does not grant any additional permissions for the cluster.

Please check if this is correct and make the necessary modifications.

File: modules/ROOT/pages/shared-tenancy.adoc [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ - delete
+ - deletecollection
+ - get
+ - list
+ - patch
+ - update
+ - watch
+----
+   
+Runtime Fabric will then create required RoleBinding objects which will use the ClusterRole rtf:agent to allow corresponding ServiceAccount to perform various actions (verbs) on cronjobs in two namespaces.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/shared-tenancy.adoc]:
```suggestion
Runtime Fabric creates the required RoleBinding objects which use the ClusterRole rtf:agent to allow corresponding ServiceAccount to perform various actions (verbs) on cronjobs in two namespaces.
```

File: modules/ROOT/pages/shared-tenancy.adoc [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+ 
+== Configure Shared Tenancy 
+
+. Create the Kubernetes objects required to install the Runtime Fabric agent:
++
+* Namespaces
+* ClusterRoles/Roles
+* RoleBindings
+* ServiceAccounts
+. Run some commands
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/shared-tenancy.adoc]: which commands?

File: modules/ROOT/pages/shared-tenancy.adoc [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  kind: ClusterRole
+  name: rtf:agent
+subjects:
+- kind: ServiceAccount
+  name: rtf-agent-sa
+  namespace: default
+----
+ 
+== Configure Shared Tenancy 
+
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/shared-tenancy.adoc]: add an intro sentence to this task?

**Review by Cristian-Venticinque — DISMISSED** [review]

File: modules/ROOT/pages/shared-tenancy.adoc (hunk 2 above)
Comment (IsaacEldridge) [reply modules/ROOT/pages/shared-tenancy.adoc]: ooof, copy/paste error, but you're right.

Comment (IsaacEldridge) [reply modules/ROOT/pages/shared-tenancy.adoc]: This is what it's supposed to say 

The following is an example of a Runtime Fabric agent ClusterRole that allows all cronjob actions (`verbs`). The ClusterRole is available in the whole cluster, but it doesn’t yet have permissions for the whole cluster.

Comment (IsaacEldridge) [inline modules/ROOT/pages/shared-tenancy.adoc]:
```suggestion
The following is an example of a Runtime Fabric agent ClusterRole that allows all cronjob actions (`verbs`). The ClusterRole is available in the whole cluster, but it doesn’t yet have permissions for the whole cluster.
```

File: modules/ROOT/pages/_partials/include-install-tasks.adoc [hunk 6]
```diff
@@ -113,3 +113,304 @@ To create the pull secret, run:
...
+  name: rtf-persistence-gateway
+  namespace: {{ .Release.Namespace }}
+- kind: ServiceAccount
+  name: cluster-status
+  namespace: {{ .Release.Namespace }}
+- kind: ServiceAccount
+  name: am-log-forwarder
+  namespace: {{ .Release.Namespace }}
+
+## ClusterRole to read underlying openshift version
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/_partials/include-install-tasks.adoc]:
```suggestion
## ClusterRole to read underlying Open Shift version
```

File: modules/ROOT/pages/_partials/include-install-tasks.adoc [hunk 7]
```diff
@@ -113,3 +113,304 @@ To create the pull secret, run:
...
+rules:
+  - apiGroups:
+      - config.openshift.io
+    resources:
+      - clusterversions
+    verbs:
+      - get
+      - list
+
+## ClusterRoleBinding for the openshift version reader role
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/_partials/include-install-tasks.adoc]:
```suggestion
## ClusterRoleBinding for the OpenShift version reader role
```

File: modules/ROOT/pages/_partials/include-install-tasks.adoc [hunk 8]
```diff
@@ -113,3 +113,304 @@ To create the pull secret, run:
...
+subjects:
+- kind: ServiceAccount
+  name: sf-edge
+  namespace: ${RTF_NAMESPACE}
+roleRef:
+  kind: ClusterRole
+  name: sf-edge-user
+  apiGroup: rbac.authorization.k8s.io
+
+## PodSecurityPoliy for persistence gateway
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/_partials/include-install-tasks.adoc]:
```suggestion
## PodSecurityPolicy for Persistence Gateway
```

File: modules/ROOT/pages/_partials/include-install-tasks.adoc [hunk 9]
```diff
@@ -113,3 +113,304 @@ To create the pull secret, run:
...
+
+* clusterRoles/Roles
+* serviceAccounts
+* clusterRoleBindings/roleBindings
+
+. In your cluster, create a file call `configuration.yml`.
+. Copy and paste the following:
++
+[source,copy]
+----
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/_partials/include-install-tasks.adoc]: @cheungaryk I need to include this very long copy snippet, as it's meant to be copy-pasted. Is there a way to collapse it when it's rendered on the docs site, so it doesn't take so much space on the page?

Comment (cheungaryk) [reply modules/ROOT/pages/_partials/include-install-tasks.adoc]: the collapse feature doesn't exist currently. If you would like to submit a feature request, please create a GUS user story for the **MS CX Engineering** product tag

**Review by emileemooreVT — DISMISSED** [review]

**Review by emileemooreVT — APPROVED** [review]

File: modules/ROOT/nav.adoc [hunk 10]
```diff
@@ -35,5 +36,4 @@
  ** xref:install-rtfctl.adoc[Use the Runtime Fabric Command Line Tool]
  ** xref:replace-license-key.adoc[Replace a Mule License Key]
  ** xref:manage-secure-properties.adoc[Manage Secure Properties with rtfctl]
- ** xref:config-mutual-auth.adoc[Configure Mutual Authentication with Anypoint Platform]
```
Comment (kmendiratta-crm) [inline modules/ROOT/nav.adoc:38]: this not needed anymore?

File: modules/ROOT/pages/_partials/include-install-tasks.adoc [hunk 11]
```diff
@@ -70,46 +62,11 @@ For GKE customers, the ingress controller included with GKE will provision a sep
...
 . Deploy an application to verify that Runtime Fabric is installed and configured correctly.
-// end::validate[]
-
-// tag::namespace[]
-
-== Create A Namespace for Runtime Fabric 
-
-You must create a namespace named `rtf` in your Kubernetes cluster. This namespace is where you install Runtime Fabric components. 
-
-To create the namespace, run:
```
Comment (kmendiratta-crm) [inline modules/ROOT/pages/_partials/include-install-tasks.adoc:93]: @IsaacEldridge @hpanchal1221 why are we removing this?

Comment (IsaacEldridge) [reply modules/ROOT/pages/_partials/include-install-tasks.adoc:93]: @kmendiratta-crm we're not removing it. this is just a reuse file and I decided not to include this there.

**Pull request merged**